src/App.tsx

```diff
@@ -1,26 +1,197 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { Options } from './types/Options';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [selectedOption, setSelectedOption] = useState<Options>(Options.All);
+  const [focus, setFocus] = useState(false);
+
+  const filterTodos = (posts: Todo[], option: Options) => {
```

Comment:
you can move this function out of the component or into another file and import it. Because now it recreates on every render